tests/unit/test_client.py

```diff
@@ -0,0 +1,84 @@
+from enum import auto
+import pytest
+from unittest.mock import Mock, patch
+from async_sendgrid.sendgrid import SendgridAPI
+from pytest_httpx import HTTPXMock
+from sendgrid.helpers.mail import Mail
+
+
+class TestAsyncClient:
+    @pytest.fixture(autouse=True)
+    def _setup(self, httpx_mock: HTTPXMock) -> None:
+        secret_key = "SECRET_KEY"
+        impersonate_subuser = "John Smith"
+        self._httpx_mock = httpx_mock
+        self._service = SendgridAPI(
+            api_key=secret_key, impersonate_subuser=impersonate_subuser
+        )
+
+    @pytest.fixture(autouse=True)
+    def _setup_data(self):
+        self._email = Mock(spec=Mail)
+        self._email.get.return_value = {}
+
+    @pytest.fixture(autouse=True)
+    def _patch_version(self):
+        with patch(
+            "async_sendgrid.sendgrid.async_sendgrid.__version__", "0.0.3"
+        ):
+            yield
```

Comment:
I haven't worked with unittest module before, why are we using 0.0.3 version here and not 1.0.0?

Reply:
Indeed we just mocked the ___version___ object over there.